### PR TITLE
Added enum constants extension

### PR DESCRIPTION
**Pull request by ryan10132**
Added extension that generates constants for enums.

### CODE REVIEW

**Comment by ryan10132** [comment]
This resolves https://github.com/vojtechhabarta/typescript-generator/issues/33

**Comment by vojtechhabarta** [comment]
@ryan10132 build of this PR failed. There is test compilation error caused by recent class rename. Can you fix it?

File: typescript-generator-core/src/test/java/cz/habarta/typescript/generator/EnumConstantsExtensionTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        Assert.assertEquals(indent + "West: <Direction>\"West\",", lines.get(4));
+        Assert.assertEquals("}", lines.get(5));
+        Assert.assertEquals("", lines.get(6));
+    }
+
+    @Test
+    public void testInTypeScriptGenerator() {
+        final Settings settings = new Settings();
+        settings.newline = "\n";
+        settings.outputFileType = TypeScriptFormat.implementationFile;
```
Comment (vojtechhabarta) [inline typescript-generator-core/src/test/java/cz/habarta/typescript/generator/EnumConstantsExtensionTest.java]: This line is not compilable, should be `TypeScriptFileType.implementationFile`.

Comment (ryan10132) [inline typescript-generator-core/src/test/java/cz/habarta/typescript/generator/EnumConstantsExtensionTest.java]: Sure, will fix later today

**Comment by ryan10132** [comment]
Thanks! When should I expect to see it in a release? More generally, how do you decide when to cut a release / how often do you do it?

**Comment by vojtechhabarta** [comment]
My current plan is to release new version each month (so next release in April).
Is that ok?

**Comment by ryan10132** [comment]
I'm a bit blocked on this feature, any chance I could get a release now? I'm not horribly blocked so I could definitely live without it.

Also I'm curious why you want to do monthly releases and not more frequently. To reduce noise? More testing / vetting? 

My understanding is that github has a "click here to release" and then enter release notes and you're done, is that right?

**Comment by vojtechhabarta** [comment]
I don't want to release too often because of several reasons. Some of them you mentioned. Next thing is that sometimes there is a combination of features which I want to release together. Improved stability when using it for some time before release. And of course it is some work: uploading artifacts to maven central repository, sometimes update documentation, release notes.

I will release it today (when you asked for it) but in general I don't plan to make frequent releases :smile:

**Comment by ryan10132** [comment]
Got it, thanks for the exception, much appreciated! :)

**Pull request merged**